streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -896,26 +904,23 @@ private void connectStateStoreNameToSourceTopicsOrPattern(final String stateStor
             }
         }
 
-
-        if (sourceTopics.isEmpty() && sourcePatterns.isEmpty()) {
-            throw new TopologyBuilderException("can't find source topic or source Pattern for state store " +
-                    stateStoreName);
+        //If both aren't empty it will force and update to the map which according to the warning above we should not do.
+        if (!sourceTopics.isEmpty() && !sourcePatterns.isEmpty()) {
```

Comment:
I agree with you. 

But my confusion stems from the comment about not updating the `stateStoreNameToSourceTopics` map once adding the state store and a collection of topics in the map.  

I think the point I'm missing is if we instead of using a pattern like `topicA*` we supplied `["topicAA", "topicAB"]` then all topics would have been associated with the state store initially.  Updating the `stateStoreNameToSourceTopics` after retrieving metadata  is actually expected when using regex.

I'll make some changes in line with your comment.